### PR TITLE
Add support for archived plugins

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add support for archived plugins

## Testing
Integration test are already using archives. However, you can test that also locally by running botkube with local plugins in the same way as you did before.

## Related issue(s)

- https://github.com/kubeshop/botkube-docs/issues/235

### CODE REVIEW

File: hack/target/build-plugins/main.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"strings"
+
+	"github.com/mattn/go-shellwords"
+
+	"github.com/kubeshop/botkube/internal/loggerx"
+)
+
+func main() {
+	pluginTargets := flag.String("plugin-targets", getEnv("PLUGIN_TARGETS", ""), "Comma separated list of specific targets to build. If empty, all targets are built.")
+	outputMode := flag.String("output-mode", getEnv("OUTPUT_MODE", "archive"), "Output format. Allowed values: binary or archive.")
```
Comment (mszostok) [inline hack/target/build-plugins/main.go:18]: in the next PR I can revert that and bring the default value to `binary`. However now the `OUTPUT_MODE` set in the workflow file is ignored so I need to do it in the code instead.

File: .github/workflows/pr-build.yaml [hunk 2]
```diff
@@ -251,3 +252,7 @@ jobs:
         run: |
           KUBECONFIG=$(k3d kubeconfig write ${{ matrix.integration }}-test-cluster) \
             make test-integration-${{ matrix.integration }}
+
+      - name: Dump cluster
```
Comment (mszostok) [inline .github/workflows/pr-build.yaml:256]: I can add the same in other places, WDYT?

**Review by pkosiec — APPROVED** [review]
LGTM, but please see the comments before merge 👍

File: .github/workflows/pr-build.yaml [hunk 3]
```diff
@@ -237,6 +237,7 @@ jobs:
         env:
           # we hardcode plugins version, so it's predictable in e2e tests
           GORELEASER_CURRENT_TAG: "v0.0.0-latest"
+          OUTPUT_MODE: "archive"
```
[the remark this answers is not in the document]
Comment (mszostok) [reply .github/workflows/pr-build.yaml:240]: my idea was to change the default as I wrote here https://github.com/kubeshop/botkube/pull/1202#discussion_r1301527837

however, even if we will not change the default it highlights that the archive is used, so I left it for now to make it more explicit, hope you don't mind 👍

**Review by pkosiec — APPROVED** [review]
👌

**Pull request merged**